runtime/android/core/src/org/xwalk/core/XWalkUpdater.java

```diff
@@ -92,17 +150,36 @@
          * update or the download (from the specified URL) is cancelled
          */
         public void onXWalkUpdateCancelled();
+
+        public void onXWalkUpdateStarted();
+        public void onXWalkUpdateFailed();
+        public void onXWalkUpdateCompleted();
     }
 
     private static final String XWALK_APK_MARKET_URL = "market://details?id=org.xwalk.core";
-    private static final String TAG = "XWalkActivity";
+
+    private static final String META_XWALK_APK_URL = "xwalk_apk_url";
+    private static final String META_XWALK_SILENT_DOWNLOAD = "xwalk_silent_download";
+
+    private static final String PRIVATE_LIB_DIR = "xwalkcorelib";
+
+    private static final String[] XWALK_LIB_RESOURCES = {
+        "libxwalkcore.so",
+        "icudtl.dat",
+        "xwalk.pak",
+        "classes.dex",
+    };
```

Comment:
How about this:

```
private static final String[] RESOURCES_TO_EXTRACT = {
    "libs/PLATFORM/libxwalkcore.so",
    "assets/icudtl.dat",
    "assets/xwalk.pak",
    "classes.dex",
};
```

We can specify the full paths here, and then extract the file name from this path, in order to avoid to check the file extensions later(not safe).